domain/credential/service/providerservice.go

```diff
@@ -87,3 +83,11 @@ func (s *WatchableProviderService) WatchCredential(ctx context.Context, key core
 	}
 	return s.st.WatchCredential(ctx, s.watcherFactory.NewNotifyWatcher, key)
 }
+
+// InvalidateCredential marks the cloud credential for the given name, cloud, owner as invalid.
+func (s *WatchableProviderService) InvalidateCredential(ctx context.Context, key corecredential.Key, reason string) error {
```

Comment:
This should live on the `ProviderService` not the `WatchableProviderService`. The watchable one is composed from the `ProviderService`.